neo/Consensus/ConsensusContext.cs

```diff
@@ -263,16 +261,7 @@ public ConsensusPayload MakeRecoveryMessage()
             PrepareRequest prepareRequestMessage = null;
             if (TransactionHashes != null)
             {
-                prepareRequestMessage = new PrepareRequest
-                {
-                    ViewNumber = ViewNumber,
-                    TransactionHashes = TransactionHashes,
-                    Nonce = Nonce,
-                    NextConsensus = NextConsensus,
-                    MinerTransaction = (MinerTransaction)Transactions[TransactionHashes[0]],
-                    Timestamp = Timestamp,
-                    ProposalStateRoot = ProposalStateRoot
-                };
```

Comment:
? Did the original code contains this?

Reply:
https://github.com/neo-project/neo/compare/master...KickSeason:fix-neox-2.x-consensus no. It's easier comparing with master, otherwise there are a mix of changes

Reply:
This is the original from `master-2.x`
```
            if (TransactionHashes != null)
            {
                prepareRequestMessage = new PrepareRequest
                {
                    ViewNumber = ViewNumber,
                    TransactionHashes = TransactionHashes,
                    Nonce = Nonce,
                    NextConsensus = NextConsensus,
                    MinerTransaction = (MinerTransaction)Transactions[TransactionHashes[0]],
                    Timestamp = Timestamp
                };
            }
```

Reply:
Right, it's 2x hahah

Reply:
But yes, this part i'ts the same as 3x, why it was changed @KickSeason ?

Reply:
Yes, it should be removed. The `StateRootSignature` of `PrepareRequest` can't be gernated by this node.

Reply:
If we want contruct `PrepareRequest`, we still need to get `StateRootSignature` from `PreparationPayloads[PrimaryIndex]`. Therefore, why not just use `PreparationPayloads[PrimaryIndex]` directly.

Reply:
It could be null

Reply:
`RecoveryMessage` support null `PrepareRequest`.
https://github.com/neo-project/neo/blob/75df7cdaeffc84cd116d907ceda11054be696af3/neo/Consensus/RecoveryMessage.cs#L138-L148